internal/gengapic/client_init.go

```diff
@@ -310,3 +311,22 @@ func generateDefaultMTLSEndpoint(defaultEndpoint string) string {
 	}
 	return defaultEndpoint
 }
+
+// generateDefaultAudience transforms a host into a an audience that can be used
+// as the `aud` claim in a JWT.
+func generateDefaultAudience(host string) string {
+	aud := host
+	// Add a scheme if not present.
+	if !strings.Contains(aud, "://") {
+		aud = "https://" + aud
+	}
+	// Remove port if present.
+	if strings.Count(aud, ":") > 1 {
+		aud = aud[:strings.LastIndex(aud, ":")]
```

Comment:
nit: How about truncating from the second colon, not from the last, in case the host includes a path with a colon? (Maybe a non-issue, but it would make this function more portable.)

Reply:
Sure, done.